### PR TITLE
VP-1423 Send customer ip to authorize.net

### PR DESCRIPTION
**Pull request by AliveMen**
Save all headers to ProcessPaymentEvaluationContext.parameters

### CODE REVIEW

File: VirtoCommerce.OrderModule.Web/Controllers/Api/OrderModuleController.cs [hunk 1]
```diff
@@ -238,12 +238,20 @@ public IHttpActionResult CalculateTotals(CustomerOrder order)
                 throw new InvalidOperationException($"Cannot find payment method with code {payment.GatewayCode}");
             }
 
+            var parameters = new NameValueCollection();
+
+            foreach (var key in HttpContext.Current.Request.Headers.AllKeys)
```
Comment (yecli) [inline VirtoCommerce.OrderModule.Web/Controllers/Api/OrderModuleController.cs]: Let's use `Keys`, as `AllKeys` is copying Keys collection, which is not required here.

Comment (AliveMen) [reply VirtoCommerce.OrderModule.Web/Controllers/Api/OrderModuleController.cs]: Keys, not a string array. Maybe better to leave AllKeys instead cast 'key' from 'Keys' collection to string?

Comment (yecli) [reply VirtoCommerce.OrderModule.Web/Controllers/Api/OrderModuleController.cs]: Agree

**Comment by yecli** [comment]
Released: https://github.com/VirtoCommerce/vc-module-order/releases/tag/v2.17.31

**Pull request merged**